apps/src/netsim/NetSimLocalClientNode.js

```diff
@@ -547,8 +551,22 @@ NetSimLocalClientNode.prototype.onWireTableChange_ = function (wireRows) {
   } else if (!mutualConnectionRow && this.myRemoteClient) {
     // Remote client disconnected or we disconnected; either way we are
     // no longer connected.
-    this.myRemoteClient = null;
-    this.remoteChange.notifyObservers(this.myWire, this.myRemoteClient);
+    this.disconnectRemote();
+  } else if (!mutualConnectionRow && ! this.myRemoteClient) {
+    // The client we're trying to connect to might have connected to
+    // someone else; check if they did and if so, stop trying to connect
+    connectingClientRow = _.find(wireRows, function(row) {
+      return row.localNodeID === this.myWire.remoteNodeID &&
+          row.remoteNodeID !== this.myWire.localNodeID;
+    }.bind(this));
+    connectingClientMutualRow = connectingClientRow ?
+        _.find(wireRows, function(row) {
+          return row.remoteNodeID == connectingClientRow.localNodeID &&
+              row.localNodeID == connectingClientRow.remoteNodeID;
+        }) : undefined;
+    if (connectingClientRow && connectingClientMutualRow) {
+      this.disconnectRemote();
+    }
```

Comment:
Kind of a naming nit, but all through here I want more distinct names to help me track what's going on.  What about:

connectingClientRow => myConnectionTarget
connectingClientMutualRow => isTargetConnectedToSomeoneElse (after making it boolean, as in last comment)